library/kani_core/src/mem.rs

```diff
@@ -338,16 +334,30 @@ macro_rules! kani_mem {
             true
         }
 
+        pub(super) mod cbmc {
+            use super::*;
+            /// CBMC specific implementation of [super::same_allocation].
+            pub fn same_allocation<T>(ptr1: *const T, ptr2: *const T) -> bool {
+                let obj1 = crate::kani::mem::pointer_object(ptr1);
+                (obj1 == crate::kani::mem::pointer_object(ptr2)) && {
+                    crate::kani::assert(
+                        unsafe {
+                            is_allocated(ptr1 as *const (), 0) || is_allocated(ptr2 as *const (), 0)
```

Comment:
I'm now confused whether we settled on `ptr1` or `obj1`.

Reply:
I prefer object since the assertion would still succeed even if both pointers are out of bounds. But I got the impression that we didn't want to do that. So I changed to assert that at least one of the pointers is valid.

Reply:
Got it